### PR TITLE
Fix underflow possibility on ColumnFamilyDbStore (#5975)

### PR DESCRIPTION
**Pull request by nyanzebra**
cherry pick of https://github.com/Azure/iotedge/pull/5975 for fixing underflow race condition that affected queue length metric

### CODE REVIEW

**Comment by nyanzebra** [comment]
Note for self: Do not merge until approval for next release

**Review by varunpuranik — APPROVED** [review]

**Pull request merged**
